### DISCUSSION TITLE
Convert to Async tree controllers

### DISCUSSION THREAD

**Discussion by blockingHD** [umbraco/Umbraco-CMS#10791] [Features and ideas]
Currently Tree Controller can't be async which causes issues when populating from other sources such as a WebAPI as HttpClientFactory uses only async methods (apart from send).

Looking at the implementation it would just be a matter of either changing the implementation of TreeController base or adding a TreeControllerAsync.

Wonder if this would be beneficial or even possible?

**Comment by nzdev** [comment]
Done :)
